### PR TITLE
Change hook_list type in netfilter hooks checking

### PR DESCRIPTION
**Pull request by v14dz**
The member hook_list of netns_nf structure is no longer a list_head since kernel version 4.9.0.  For kernel versions from 4.9.0 to 4.14.0 change this member type to nf_hook_entry.  This commit makes Tyton work under Debian 9.

### CODE REVIEW

File: src/netfilter_hooks.c [hunk 1]
```diff
@@ -64,25 +64,59 @@ static void search_hooks(const struct nf_hook_entries *e){
...
 	unsigned long addr;
 	struct module *mod;
 	struct nf_hook_ops *elem;
+	struct nf_hook_entry *cur_hook_entry = NULL;
 
-	list_for_each_entry(elem, hook_list, list){
+	cur_hook_entry = (struct nf_hook_entry *) hook_list;
+
+	while(cur_hook_entry) {
+
```
Comment (nbulischeck) [inline src/netfilter_hooks.c:80]: Nits: I prefer to have no newline after a while loop, as well as putting the curly brace against the right paren and putting a space after `while` and before the left paren.

**Comment by nbulischeck** [comment]
According to the build logs, this is failing for the 4.4.0-101-generic kernel.

**Comment by v14dz** [comment]
Yes, I realized that I missed some KERNEL_VERSION macros to keep compatibility with 4.4.0.

**Comment by v14dz** [comment]
I rebased to have a single commit (w/ newline removed and additional macros for checking versions). This commit should also fix https://github.com/nbulischeck/tyton/issues/24 (the kernel version corresponds to Debian 9).

**Comment by nbulischeck** [comment]
I saw that! Much appreciated 😄

**Pull request merged**